src/main/java/edu/harvard/iq/dataverse/ExternalVocabularyValue.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+
+package edu.harvard.iq.dataverse;
+
+import edu.harvard.iq.dataverse.util.BundleUtil;
+import org.apache.commons.lang3.StringUtils;
+
+import java.io.Serializable;
+import java.sql.Timestamp;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.Objects;
+import java.util.MissingResourceException;
+
+import javax.json.JsonObject;
+import javax.persistence.CascadeType;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.Index;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.OneToMany;
+import javax.persistence.Table;
+
+/**
+ *
+ * @author qqmyers
+ */
+@Entity
+@Table(indexes = { @Index(columnList = "datasetfieldtype_id"), @Index(columnList = "displayorder") })
+public class ExternalVocabularyValue implements Serializable {
+
+    public ExternalVocabularyValue() {
+    }
+
+    public ExternalVocabularyValue(String uri, String value) {
+        this.uri = uri;
+        this.value = value;
+    }
+
+    @Id
+    @Column(columnDefinition = "TEXT", nullable = false)
+    private String uri;
```

Comment:
As of now, I haven't made this change. I think it could be changed that way if the uri is marked as being required to be unique (which makes JPA add a unique constraint for this column when first creating the table). That said, there are no queries except finding an entry by uri, so adding the id and making it the primary key doesn't seem to buy much except conformity. If this is something IQSS wants, let me know and I'll make the change and verify that it keeps the uri unique.

Reply:
I defer to @scolapasta on the consistency of using `id`.

Reply:
@qqmyers @pdurbin I think we should stay consistent - generally the pattern is for the id to not contains any info.

Reply:
OK - hopefully the last commit does it - just adding the generated id column and adding a unique constraint on the uri.